src/runtime/Memory.h

```diff
@@ -39,13 +39,24 @@ class Memory : public Extern {
     // Caching memory target for fast access.
     struct TargetBuffer {
         TargetBuffer()
-            : prev(nullptr)
-            , buffer(nullptr)
-            , sizeInByte(0)
         {
+            setUninitialized();
         }
 
-        TargetBuffer* prev;
+        static size_t sizeInPointers(size_t memoryCount)
+        {
+            return ((memoryCount * sizeof(Memory::TargetBuffer)) + (sizeof(void*) - 1)) / sizeof(void*);
+        }
+
+        void setUninitialized()
+        {
+            sizeInByte = ~(uint64_t)0;
+        }
+
+        void enque(Memory* memory);
+        void deque(Memory* memory);
+
+        TargetBuffer* next;
```

Comment:
Is there any specific purpose of this next reference?
It seems that `TargetBuffer` can be linked to each other like linked-list,
but is it possible that one Memory has multiple `TargetBuffers`?

Reply:
Yes. Each Instance which has a reference to a given Memory has a TargetBuffer connected to that memory. The TargetBuffer is simply a cache, since memory properties rarely changes, and a key resource (memory accesses are very frequent).